File: browser-dm/src/main/scala/com/xyz/browser/dm/entity/Block.scala
package com.xyz.browser.dm.entity

case class Block(
//                  jsonrpc: String,
//                  id: Long,
                  difficulty:String,
                  extraData:String,
                  gasLimit:String,
                  gasUsed:String,
                  hash:String,
                  logsBloom:String,
                  miner:String,
                  mixHash:String,
                  nonce:String,
                  number:String,
                  parentHash:String,
                  receiptsRoot:String,
                  sha3Uncles:String,
                  size:String,
                  stateRoot:String,
                  timestamp:String,
                  totalDifficulty:String,
                  transactions:String,//jsonStr arr
                  transactionsRoot: String,
                  uncles:String//jsonStr arr
                )extends Record

File: browser-app/src/main/java/com/xyz/browser/app/modular/hbase/model/Block.java
package com.xyz.browser.app.modular.hbase.model;

import com.baomidou.mybatisplus.activerecord.Model;
import com.baomidou.mybatisplus.annotations.TableField;
import com.baomidou.mybatisplus.annotations.TableId;
import com.baomidou.mybatisplus.annotations.TableName;
import lombok.Data;

import java.io.Serializable;

@Data
@TableName("block")
public class Block extends Model<Block> {
    @TableField("difficulty")
    private String difficulty;
    @TableField("extraData")
    private String extraData;
    @TableField("gasLimit")
    private String gasLimit;
    @TableField("gasUsed")
    private String gasUsed;
    @TableId(value = "hash")
    private String hash;
    @TableField("logsBloom")
    private String logsBloom;
    @TableField("miner")
    private String miner;
    @TableField("mixHash")
    private String mixHash;
    @TableField("nonce")
    private String nonce;
    @TableField("number")
    private String number;
    @TableField("parentHash")
    private String parentHash;
    @TableField("receiptsRoot")
    private String receiptsRoot;
    @TableField("sha3Uncles")
    private String sha3Uncles;
    @TableField("size")
    private String size;
    @TableField("stateRoot")
    private String stateRoot;
    @TableField("timestamp")
    private String timestamp;
    @TableField("totalDifficulty")
    private String totalDifficulty;
    @TableField("transactions")
    private String transactions;//jsonStr arr
    @TableField("transactionsRoot")
    private String transactionsRoot;
    @TableField("uncles")
    private String uncles;//jsonStr arr

    @Override
    protected Serializable pkVal() {
        return this.hash;
    }
}
